wallet/ledger/ledger.go

```diff
@@ -87,14 +86,50 @@ func (af *ledgerAccountFactory) SupportedImportKinds() []wallet.ImportKind {
 }
 
 func (af *ledgerAccountFactory) HasConsensusSigner(rawCfg map[string]interface{}) bool {
-	return true
+	cfg, err := af.unmarshalConfig(rawCfg)
+	if err != nil {
+		return false
+	}
+
+	switch cfg.Algorithm {
+	case wallet.AlgorithmEd25519Legacy, wallet.AlgorithmEd25519Adr8:
+		return true
+	}
+	return false
+}
+
+// migrate migrates the given config ledger account entry to the latest version of the config and
+// returns true, if any changes were needed.
+func (af *ledgerAccountFactory) migrate(raw map[string]interface{}) bool {
+	var changed bool
+
+	// CONFIG MIGRATION 1 (add legacy derivation support): Set default derivation to ADR 8, if not set.
+	if raw["derivation"] == nil && raw["algorithm"] == nil {
+		raw["derivation"] = "adr8"
+
+		changed = true
+	}
+
+	// CONFIG MIGRATION 2 (convert derivation -> algorithm).
+	if val, ok := raw["derivation"]; ok {
+		raw["algorithm"] = fmt.Sprintf("ed25519-%s", val)
+		delete(raw, "derivation")
+
+		changed = true
+	}
+
+	return changed
 }
 
 func (af *ledgerAccountFactory) unmarshalConfig(raw map[string]interface{}) (*accountConfig, error) {
 	if raw == nil {
 		return nil, fmt.Errorf("missing configuration")
 	}
 
+	if changed := af.migrate(raw); changed {
+		config.Global().Save()
```

Comment:
I worry this could have unintended side effects as there could be other configuration changes pending that should not be saved. Could we make a separate `Migrate(name, oldCfg) newCfg` step on the factory which would perform any migrations on the accounts? And then we could have a migration step in the `initConfig` phase (see `cmd/root.go`).

Reply:
I did something similar now. Can you check it out?